jwst/extract_2d/grisms.py

```diff
@@ -130,12 +130,18 @@ def extract_tso_object(
     output_model.update(input_model)
     subwcs = copy.deepcopy(input_model.meta.wcs)
 
+    # (Some) NIRCam wavelengthrange entries have a fieldpoint entry, while NIRISS does not.
+    if len(wavelengthrange[0]) == 5:
+        filter_idx = 2
+    else:
+        filter_idx = 1
```

Comment:
I understand why you did it this way and I don't see a way around it, but I still don't like this.  It bakes in so many assumptions about the structure of those models.  Is there a way to request that this model structure changes to be a Table with column labels or something?

And/or, per [Brett's comment](https://github.com/spacetelescope/stdatamodels/issues/516#issuecomment-3020119877), somehow ensure that the people delivering the reference files are forced to attempt validation, and request that we make any changes needed before the reffile delivery deadline instead of the Monday after the delivery was due?

Did you test that this still works on an old CRDS context?  I'm seeing that the `else` part of the statement is not covered by unit tests.

Reply:
I understand your motivations - the built-in assumptions are fragile. But this PR is intended to get unit tests passing and to ease the patch delivery process. We can attempt to make the wavelengthrange file and its use in the pipeline more robust as part of work for JP-3990 in B12.1. The issues we ran into here were due to the mismatched timing between CRDS work and JP work for this update, and B12.0(.1) needs to work with the existing CRDS context.